fcrepo-kernel-impl/src/test/java/org/fcrepo/kernel/impl/TransactionImplTest.java

```diff
@@ -0,0 +1,173 @@
+        when(txManager.getPersistentStorageSessionManager()).thenReturn(pssManager);
+        testTx = new TransactionImpl("123", txManager);
+    }
+
+    @Test
+    public void testGetId() {
+        assertEquals("123", testTx.getId());
+    }
+
+    @Test
+    public void testDefaultShortLived() {
+        assertEquals(true, testTx.isShortLived());
+    }
+
+    @Test
+    public void testSetShortLived() {
+        testTx.setShortLived(false);
+        assertEquals(false, testTx.isShortLived());
+    }
+
+    @Test
+    public void testCommit() {
+        testTx.commit();
+        try {
+            verify(psSession).commit();
+        } catch (PersistentStorageException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void testCommitIfShortLived() {
+        testTx.setShortLived(true);
+        testTx.commitIfShortLived();
+        try {
+            verify(psSession).commit();
+        } catch (PersistentStorageException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void testCommitIfShortLivedOnNonShortLived() {
+        testTx.setShortLived(false);
+        testTx.commitIfShortLived();
+        try {
+            verify(psSession, never()).commit();
+        } catch (PersistentStorageException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void testCommitExpired() {
+        testTx.expire();
+        testTx.commit();
+        try {
+            verify(psSession).commit();
+        } catch (PersistentStorageException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void testCommitRolledbackTx() {
+        testTx.rollback();
+        testTx.commit();
+        try {
+            verify(psSession, never()).commit();
+        } catch (PersistentStorageException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void testRollback() {
+        testTx.rollback();
+        try {
+            verify(psSession).rollback();
+        } catch (PersistentStorageException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void testRollbackCommited() {
+        testTx.commit();
+        testTx.rollback();
+        try {
+            verify(psSession, never()).rollback();
+        } catch (PersistentStorageException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void testExpire() {
+        testTx.expire();
+        assertEquals(true, testTx.hasExpired());
+    }
+
+    @Test
+    public void testRefresh() {
+        final Instant previousExpiry = testTx.getExpires();
+        testTx.refresh();
+        assertTrue(testTx.getExpires().isAfter(previousExpiry));
+    }
+
+    @Test
+    public void testNewTransactionNotExpired() {
+        assertTrue(testTx.getExpires().compareTo(Instant.now()) > 0);
```

Comment:
For readability/consistency, could this be
`assertFalse(testTx.getExpires().isAfter(previousExpiry));`
instead?